simulations/src/node/mod.rs

```diff
@@ -201,3 +201,189 @@ impl NodeIdExt for NodeId {
         NodeId::new(bytes)
     }
 }
+
+pub trait ResearchNodeIdExt: NodeIdExt {
+    fn parent_index(&self) -> usize;
+}
+
+impl ResearchNodeIdExt for NodeId {
+    fn parent_index(&self) -> usize {
+        const SIZE: usize = core::mem::size_of::<usize>();
+        let x: [u8; 32] = (*self).into();
+        let mut buf = [0; SIZE];
+        buf.copy_from_slice(&x[32 - SIZE..]);
+        usize::from_be_bytes(buf)
+    }
+}
+
+/// Reference https://github.com/logos-co/nomos-specs/blob/Carnot-Simulation/carnot/carnot_simulation_psuedocode.py
+pub trait CommitteeExt {
```

Comment:
Not sure we do really want to do things like this actually. What does this offer that we cannot really simulate with a `TreeBranch` overlay?

Reply:
It should be how to construct the Committee, in specs, it seems that the committee is built based on total levels. I have a new idea about wrap the TreeOverlay and make it has a different construct method follows the specs.

Reply:
I would really like to discuss this. Those specs were not approved and are just a hint actually.

Reply:
Yeah, here is a more detailed explanation https://www.notion.so/Carnot-Simulation-Mechanism-c025dbab6b374c139004aae45831cf78. Basically, I am also not sure do we want to fully follow their thoughts?

Reply:
We do. But, imo, we just need a single branch overlay, nothing else. Everything else that we need is already provided by the application features.

Reply:
Could I know the meaning of the single branch overlay?

Reply:
![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/ecb9e787-bbbd-4cd9-9e76-3e50771d935e/Untitled.png)

That style of overlay

Reply:
Cool, thanks. In our rust implementation, the committee contains a set of nodes. However, I wonder what the numbers in the above picture(1, 2, 4, 7) refer to. Is the (1, 2, 4, 7) together a committee, or does the single leaf, such as 1, indicate one committee?

Reply:
Forget about the numbers. Numbers are the simple committee ids from the real tree representation (which is much larger than just a branch).
We need:
```
RootCommittee <=> RootCommitteeChild <=> Committee A <=> ........ <=> Committee Z
```